src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -40,40 +42,69 @@ public static boolean isInformative(final double[] gls) {
     }
 
     /**
-     * Creates a VariantContextWriter who's outputFile type is <code>VariantContextWriterBuilder.OutputType.VCF</code>.
+     * Creates a VariantContextWriter who's outputFile type is based on the extension of the output file name.
      * The default options set by VariantContextWriter are cleared before applying ALLOW_MISSING_FIELDS_IN_HEADER (if
      * <code>lenientProcessing</code> is set), followed by the set of options specified by any <code>options</code> args.
      *
-     * @param outFile output File for this writer
+     * @param outFile output File for this writer. May not be null.
      * @param referenceDictionary required if on the fly indexing is set, otherwise can be null
-     * @param lenientProcessing true if ALLOW_MISSING_FIELDS_IN_HEADER option should be set
-     * @param options variable length list of additional options to be set for this writer
+     * @param createMD5 true if an md5 file should be created
+     * @param options variable length list of additional Options to be set for this writer
      * @returns VariantContextWriter must be closed by the caller
      */
     public static VariantContextWriter createVCFWriter(
             final File outFile,
             final SAMSequenceDictionary referenceDictionary,
-            final boolean lenientProcessing,
+            final boolean createMD5,
             final Options... options)
     {
-        VariantContextWriterBuilder vcWriterBuilder = new VariantContextWriterBuilder()
-                                                            .clearOptions()
-                                                            .setOutputFile(outFile)
-                                                            .setOutputFileType(VariantContextWriterBuilder.OutputType.VCF);
+        Utils.nonNull(outFile);
+
+        VariantContextWriterBuilder vcWriterBuilder =
+                new VariantContextWriterBuilder().clearOptions().setOutputFile(outFile);
+
+        if (VariantContextWriterBuilder.OutputType.UNSPECIFIED == getVariantFileTypeFromExtension(outFile)) {
+            // the only way the user has to specify an output type is by file extension, and htsjdk
+            // throws if it can't map the file extension to a known vcf type, so fallback to a default
+            // of VCF
+            logger.warn(String.format(
+                    "Can't determine output variant file format from output file extension \"%s\". Defaulting to VCF.",
+                    FilenameUtils.getExtension(outFile.getPath())));
+            vcWriterBuilder = vcWriterBuilder.setOutputFileType(VariantContextWriterBuilder.OutputType.VCF);
```

Comment:
No, VariantContextWriterBuilder will derive the output type itself based on the file extension, its just that its not very forgiving and throws if it doesn't recognize they extension as a known type (i.e, a name that ends in .tmp). Perhaps I shouldn't be trying to do this at all, and we should just let it throw if such cases. Let me know what you think.